_source/_posts/2020-10-04-introducing-okta-cli.md

```diff
@@ -0,0 +1,319 @@
+Company: Okta
+Creating new Okta Organization, this may take a minute:
+OrgUrl: https://dev-159750.okta.com
+An email has been sent to you with a verification code.
+
+Check your email
+Verification code: 189919
+New Okta Account created!
+Your Okta Domain: https://dev-159750.okta.com
+To set your password open this link:
+https://dev-159750.okta.com/reset_password/drpdRFTmBi8yQ-uIzbpx?fromURI=/
+
+Select a sample
+> 1: Spring Boot + Okta
+> 2: Vue + Okta
+> 3: ASP.net Core 3  MVC
+> 4: Angular + Okta
+Enter your choice [Spring Boot + Okta]: 3
+Configuring a new OIDC Application, almost done:
+Created OIDC application, client-id: 0oa6kobyoZO41crqQ5d5
+
+Change the directory:
+    cd okta-aspnet-core3-sample
+
+Okta configuration written to: okta-aspnetcore-mvc-example/appsettings.json
+Don't EVER commit okta-aspnetcore-mvc-example/appsettings.json into source control
+Run this application with:
+    dotnet run --project okta-aspnetcore-mvc-example
+```
+
+There's a lot going on here from the humble `okta start` command, so let's break it down.
+
+First I see a one-time requirement in order to provision the Okta org.
+
+I put in some minimal information: name, email and company. In about 20 seconds, a shiny new Okta org is allocated for me and the Okta CLI shows me my Org URL (https://dev-159750.okta.com).
+
+Next, I need to prove that I own the email address I entered. an One-Time Passcode (OTP) is sent to my email address and the Okta CLI waits for me to paste that in.
+
+{% img blog/introducing-okta-cli/okta-email.png alt:"okta email" width:"600" %}{: .center-image }
+
+Once my email address is verified, the Okta CLI displays a unique link to set my password.
+
+> **NOTE:** The order-of-operations here is not important. You can set your password before you choose a sample or afterwards. But, you should set your password in a timely fashion as that unique URL expires after 24 hours.
+
+Next, I see a list of available samples. As of this writing, there are four samples, but we are expanding that list. And, the Okta CLI automatically makes any new samples available in the list. You can create as many sample as you like with your Okta Org. I chose option 3: `ASP.net Core 4 MVC`
+
+Finally, I see information about the Application that was provisioned in my Okta Org along with instructions to run the sample app.
+
+I'm going to follow those instructions and run this sample:
+
+```bash
+cd okta-aspnet-core3-sample
+dotnet run --project okta-aspnetcore-mvc-example
+```
+
+Here's the output (abbreviated):
+
+```bash
+dotnet run --project okta-aspnetcore-mvc-example
+
+Welcome to .NET Core 3.1!
+---------------------
+SDK Version: 3.1.402
+
+...
+
+info: Microsoft.Hosting.Lifetime[0]
+      Now listening on: https://localhost:5001
+info: Microsoft.Hosting.Lifetime[0]
+      Application started. Press Ctrl+C to shut down.
+...
+```
+
+> Did you know you could run dotnet on macOS? What a world! Simply run: `brew cask install dotnet-sdk`
+
```

Comment:
Maybe mention a private/incognito window, to make it more obvious the user has logged in.